drivers/amazonec2/amazonec2.go

```diff
@@ -250,7 +250,7 @@ func (d *Driver) Create() error {
 		}
 
 		for _, s := range subnets {
-			if s.AvailabilityZone == regionZone {
+			if s.AvailabilityZone == regionZone && s.VpcId == d.VpcId {
```

Comment:
It might also be reasonable to see if one of the subnets is the default for the AZ and prefer that one.